app/src/main/java/ch/epfl/favo/view/tabs/MapPage.java

```diff
@@ -352,15 +372,35 @@ private void onToggleClick(View view) {
     Navigation.findNavController(requireView()).navigate(R.id.action_nav_map_to_nearby_favor_list);
   }
 
+  private void onLookThroughClick(View view) {
+    for (Marker marker : favorsAroundMe.values()) {
+      ArrayList tagArray = (ArrayList) marker.getTag();
+      boolean visited = (boolean) tagArray.get(2);
+      if (!visited) {
+        tagArray.set(2, true);
+        marker.showInfoWindow();
+        mMap.animateCamera(
+            CameraUpdateFactory.newLatLngZoom(marker.getPosition(), defaultZoomLevel + 2));
+        return;
+      }
+    }
+    // if all visited
+    for (Marker marker : favorsAroundMe.values()) {
+      ArrayList tagArray = (ArrayList) marker.getTag();
+      tagArray.set(2, false);
+    }
+    Toast.makeText(requireContext(), "you have visited all nearby favors", Toast.LENGTH_SHORT)
```

Comment:
I would extract the string in the strings.xml

Reply:
fixed